contracts/libraries/Assessment/AssessmentVoteLib.sol

```diff
@@ -0,0 +1,176 @@
+    }
+    IAssessment.IncidentDetails memory incidentDetails = incidents[id].details;
+    uint payoutImpact = AssessmentUtilsLib._getPayoutImpactOfIncident(incidentDetails);
+    return payoutImpact * CONFIG.REWARD_PERC / PERC_BASIS_POINTS;
+  }
+
+  // [todo] Expose a view to find out the last index until withdrawals can be made and also
+  //  views for total rewards and withdrawable rewards
+  function withdrawReward (
+    IAssessment.Configuration calldata CONFIG,
+    INXMToken nxm,
+    address user,
+    uint104 untilIndex,
+    mapping(address => IAssessment.Stake) storage stakeOf,
+    mapping(address => IAssessment.Vote[]) storage votesOf,
+    IAssessment.Claim[] storage claims,
+    IAssessment.Incident[] storage incidents
+  ) external returns (uint rewardToWithdraw, uint104 withdrawUntilIndex) {
+    IAssessment.Stake memory stake = stakeOf[user];
+    {
+      uint voteCount = votesOf[user].length;
+      withdrawUntilIndex = untilIndex > 0 ? untilIndex : uint104(voteCount);
+      require(
+        untilIndex <= voteCount,
+        "Vote count is smaller that the provided untilIndex"
+      );
+      require(stake.rewardsWithdrawnUntilIndex < voteCount, "No withdrawable rewards");
+    }
+
+    uint totalReward;
+    for (uint i = stake.rewardsWithdrawnUntilIndex; i < withdrawUntilIndex; i++) {
+      IAssessment.Vote memory vote = votesOf[user][i];
+      require(
+        block.timestamp > AssessmentUtilsLib._getVoteLockupEndDate(CONFIG, vote),
+        "Cannot withdraw rewards from votes which are in lockup period"
+      );
+      IAssessment.Poll memory poll =
+        IAssessment.EventType(vote.eventType) == IAssessment.EventType.CLAIM
+        ? claims[vote.eventId].poll
+        : incidents[vote.eventId].poll;
+
+      totalReward = _getTotalRewardForEvent(
+        CONFIG,
+        IAssessment.EventType(vote.eventType),
+        vote.eventId,
+        claims,
+        incidents
+      );
+      rewardToWithdraw += totalReward * vote.tokenWeight / (poll.accepted + poll.denied);
+    }
+
+    stakeOf[user].rewardsWithdrawnUntilIndex = withdrawUntilIndex;
+    nxm.mint(user, rewardToWithdraw);
+  }
+
+  function castVote (
+    IAssessment.Configuration calldata CONFIG,
+    uint8 eventType,
+    uint104 id,
+    bool accepted,
+    mapping(address => IAssessment.Stake) storage stakeOf,
+    mapping(address => IAssessment.Vote[]) storage votesOf,
+    mapping(bytes32 => bool) storage hasAlreadyVotedOn,
+    IAssessment.Claim[] storage claims,
+    IAssessment.Incident[] storage incidents
+  ) external {
+
+    {
+      bytes32 voteHash = keccak256(abi.encodePacked(id, msg.sender, eventType));
+      require(!hasAlreadyVotedOn[voteHash], "Already voted");
+      hasAlreadyVotedOn[voteHash] = true;
+    }
+
+    IAssessment.Stake memory stake = stakeOf[msg.sender];
+    require(stake.amount > 0, "A stake is required to cast votes");
+
+    uint payoutImpact;
```

Comment:
what is a payoutImpact ? is it just the amount to be paid out? name is a bit confusing

Reply:
You got it right. When talking about claims it is the requested amount, when talking about incidents it is an estimate of total possible payouts determined by a fixed percentage (probably 30%) and the active cover amount. The two are unified under the term payout impact which best describes in either of the two cases the impact the payouts would have on the pool, hence the importance of the claim/incident. Large payout impacts means the polls are likely to get extended to the maximum period as they require higher participation.